Standardize/Add missing docblocks

rafaelgallani opened the pull request:
This PR standardizes the doc blocks for all of our methods and adds a few that were missing. 

## Testing
Inspect the generated docs and make sure it's all good 😬

rafaelgallani committed:
Add missing docblocks and standardize wording

Standardized:

- Methods that were related to the `Iterator` interface
- Constructor methods - `Entity`, `Command`
- `@internal` tag enclosed with empty lines

rafaelgallani committed:
Standardize constructors, summaries, and descriptions

rafaelgallani committed:
Update docs

rafaelgallani committed:
Update docs - based on phpdoc-markdown repo unmerged PR

rafaelgallani commented on composer.json:
```diff
@@ -34,5 +34,11 @@
+      "url": "https://github.com/Automattic/phpdoc-markdown.git"
+    }
+  ]
 }
```
This should also be removed once https://github.com/Automattic/phpdoc-markdown/pull/2 gets merged.

rafaelgallani commented on composer.json:
```diff
@@ -23,7 +23,7 @@
     "http-interop/http-factory-guzzle": "^1.0",
     "php-http/mock-client": "^1.0",
-    "automattic/phpdoc-markdown": "^1.0"
+    "automattic/phpdoc-markdown": "dev-add/remove-empty-references"
```
This is temporary. 

It depends on a new version that should be released once https://github.com/Automattic/phpdoc-markdown/pull/2 gets merged.

In short: we're now removing empty interfaces, traits, and namespaces from the documentation.

ramynasr commented on lib/configuration.php:
```diff
@@ -18,11 +18,29 @@
+ * Configuration class for the API settings.
+ */
 class Configuration {
+	// Formats for the query string boolean values
```
This one won't be picked up by phpdoc.

ramynasr commented on lib/entity/domain-contact.php:
```diff
@@ -47,6 +47,13 @@ class Domain_Contact {
+	 *
+	 * @param \Automattic\Domain_Services\Entity\Contact_Id|null $contact_id
+	 * @param \Automattic\Domain_Services\Entity\Contact_Information|null $contact_info
+	 * @param \Automattic\Domain_Services\Entity\Contact_Disclosure|null $disclose_fields
```
I think we can remove `\Automattic\Domain_Services\Entity\` from these since that's the namespace we are in.

ramynasr commented on lib/entity/domain-contacts.php:
```diff
@@ -253,37 +255,56 @@ public function set_billing( Domain_Contact $billing ): void {
 	 * @internal
+	 *
+	 * @return Domain_Contact|null
+	 * @throws \Automattic\Domain_Services\Exception\Entity\Invalid_Value_Exception
```
We already import `Exception` above, so probably `Exception\Entity\Invalid_Value_Exception` is shorter and easier to scan/read

ramynasr reviewed (approved):
Thank you so much for going through all of this. I love the changes you introduced here!!

Just a nitpick or two.

rafaelgallani committed:
Add action to check for broken markdown links

rafaelgallani committed:
Replace `@see` target with `Code` class instead of `Code::Description`

rafaelgallani committed:
Add better comments for phpdoc

rafaelgallani committed:
Simplify local and already imported namespaces

rafaelgallani committed:
Update docs (again 😬)

rafaelgallani commented:
☝🏻 Rebase on `trunk` and updating docs with its latest changes

rafaelgallani committed:
Include interfaces with visible constants

Newest plugin version also includes interfaces with visible (depending on phpdoc.xml configuration) constants. This commit adds the classes that implement these interfaces.

rafaelgallani committed:
Revert composer package - dependency new version was released

rafaelgallani committed:
Generate docs with renamed namespace

rafaelgallani commented:
After rebasing with `trunk`, some files are showing as new instead of renamed. For some reason, git doesn't think they're similar enough, but it's the same file. Sorry for that.

rafaelgallani commented:
Replaced by #162 - lots of conflicting files and non-related changes.
